integration/yarn/src/main/java/alluxio/yarn/ApplicationMaster.java

```diff
@@ -397,6 +410,39 @@ private void launchWorkerContainer(Container container) {
     }
   }
 
+  /**
+   * Checks if an Alluxio master node is already running
+   * or not on the master address given.
+   *
+   * @return true if master exists, false otherwise
+   */
+  private boolean masterExists() {
+
+    String webPort = Configuration.get(PropertyKey.MASTER_WEB_PORT);
+
+    try {
+      URL myURL = new URL("http://" + mMasterAddress + ":" + webPort + Constants.REST_API_PREFIX + "/master/version");
+      LOG.debug("Checking for master at: " + myURL.toString());
+      HttpURLConnection connection = (HttpURLConnection) myURL.openConnection();
+      connection.setRequestMethod(HttpMethod.GET);
+      int resCode = connection.getResponseCode();
+      LOG.debug("Response code from master was: " + Integer.toString(resCode));
+
+      if (resCode == HttpURLConnection.HTTP_OK) {
+        connection.disconnect();
+        return true;
+      } else {
+        connection.disconnect();
+      }
+    } catch (MalformedURLException e) {
+      LOG.error("Malformed URL in attempt to check if master is running already", e);
+    } catch (IOException e) {
+      LOG.debug("Unable to open connection to master in attempt to check if already running", e);
```

Comment:
No, we expect it to fail if a Master Node is not already running.
I can word the error message a little different as it seems to infer an error.